deployment/conf/.templates/setenv.sh.templ

```diff
@@ -1,17 +1,15 @@
 #!/bin/sh
-#
-
 CATALINA_HOME=/usr/share/tomcat8
 
 # The first existing directory is used for JAVA_HOME (if JAVA_HOME is not
 # defined in $DEFAULT)
-JDK_DIRS="/usr/lib/jvm/java-8-openjdk-amd64 /usr/lib/jvm/jre-8-oracle-x64"
+JDK_DIRS="/usr/lib/jvm/java-11-openjdk-amd64 /usr/lib/jvm/java-8-openjdk-amd64"
```

Comment:
updating for Java 11

Reply:
Shouldn't the 2nd dir also be java 11?

Reply:
Shouldn't the 2nd dir be `/usr/lib/jvm/jre-11-openjdk-x64`? Let's just ditch 8 completely if we're moving to 11

Reply:
I was in several minds about what to do with this script (template). The existing script is a modification of a `setenv.sh` script installed as part of tomcat, and the current version changes the tomcat script by altering the JDK directories to look for JDK 8 and adding in run time parameters for tomcat.

original tomcat script:
```sh
#!/bin/sh
#

CATALINA_HOME=/usr/share/tomcat8

# The first existing directory is used for JAVA_HOME (if JAVA_HOME is not
# defined in $DEFAULT)
JDK_DIRS="/usr/lib/jvm/java-7-openjdk-amd64 /usr/lib/jvm/jre-7-oracle-x64"

# Look for the right JVM to use
for jdir in $JDK_DIRS; do
    if [ -r "$jdir/bin/java" -a -z "${JAVA_HOME}" ]; then
        JAVA_HOME_TMP="$jdir"
        # checks for a real JDK like environment, needed to check if 
        # really the java-gcj-compat-dev package is installed
        if [ -r "$jdir/bin/jdb" ]; then
            JAVA_HOME="$JAVA_HOME_TMP"
        fi
    fi
done

# Default Java options
if [ -z "$JAVA_OPTS" ]; then
        JAVA_OPTS="-Djava.awt.headless=true"
fi
```
(n.b. `JAVA_HOME` is already set to `/usr/local/openjdk-8`; neither of the paths in `JDK_DIRS` exist)

version that appears on the current workspace image:
```sh
#!/bin/sh
#

# CATALINA_HOME=/usr/share/tomcat8

# The first existing directory is used for JAVA_HOME (if JAVA_HOME is not
# defined in $DEFAULT)
JDK_DIRS="/usr/lib/jvm/java-8-openjdk-amd64 /usr/lib/jvm/jre-8-oracle-x64"

# Look for the right JVM to use
for jdir in $JDK_DIRS; do
    if [ -r "$jdir/bin/java" -a -z "${JAVA_HOME}" ]; then
        JAVA_HOME_TMP="$jdir"
        # checks for a real JDK like environment, needed to check if
        # really the java-gcj-compat-dev package is installed
        if [ -r "$jdir/bin/jdb" ]; then
            JAVA_HOME="$JAVA_HOME_TMP"
        fi
    fi
done

# Default Java options
if [ -z "$JAVA_OPTS" ]; then
        JAVA_OPTS="-Djava.awt.headless=true -server \
               -Xms1000m -Xmx3000m \
               -XX:+UseG1GC"
fi
```
(note the extra `JAVA_OPTS`)

Even if `JAVA_HOME` was not already set, it wouldn't be changed by this script as there's no `jdb` in the bin directory. I don't know why this content was kept in the script, other than perhaps inertia.

The `setenv.sh` script thus could/should be reduced to these lines:

```sh
# Default Java options
if [ -z "$JAVA_OPTS" ]; then
        JAVA_OPTS="-Djava.awt.headless=true -server \
               -Xms1000m -Xmx3000m \
               -XX:+UseG1GC"
fi
```

Reply:
If I understand this correctly: If JAVA_HOME and JAVA_OPTS are non-empty strings, this script does nothing (I hate shell script syntax...)?

If so, IMO we should not monkey patch scripts installed by apt-get and instead just set JAVA_HOME and JAVA_OPTS in the docker file, maybe by running a script that sets them based on environment variables as the current script does.

That being said, that might be too much work for this PR. At some point I want to go through the entire dockerfile to update tomcat and simplify the installation and startup as much as possible, but maybe not here

Reply:
That being said, we still shouldn't have the references to java 8 in this script, whatever we do

Reply:
I've removed all but the lines that actually do something, i.e. the lines that set the `JAVA_OPTS`. The tomcat docs suggest customising `setenv.sh` for your environment/needs (see https://tomcat.apache.org/tomcat-8.5-doc/RUNNING.txt) so there's no point in keeping cruft around.

Reply:
:+1: